src/job.rs

```diff
@@ -251,12 +249,13 @@ impl Job {
 
         smtp.connectivity.set_working(context).await;
 
-        let status = match smtp.send(context, recipients, message, job_id).await {
+        let send_result = smtp.send(context, recipients, message, job_id).await;
+        smtp.last_send_error = send_result.as_ref().err().map(|e| e.to_string());
+
+        let status = match send_result {
             Err(crate::smtp::send::Error::SmtpSend(err)) => {
                 // Remote error, retry later.
                 warn!(context, "SMTP failed to send: {:?}", &err);
-                smtp.connectivity.set_err(context, &err).await;
-                self.pending_error = Some(err.to_string());
```

Comment:
Before, only `Error::SmtpSend` let to the connectivity being set correctly. Now, it is set for every error.